Clarify what settings are not respected by dotnet restore (#13514)
dotnet/docs: docs/core/tools/dotnet-restore.md

For project-specific tooling, `dotnet restore` first restores the package in which the tool is packed, and then proceeds to restore the tool's dependencies as specified in its project file.

> ### Nuget.Config differences

Reviewer: ```suggestion
### Nuget.Config differences

```

The behavior of the `dotnet restore` command is affected by the settings in the *Nuget.Config* file, if present. For example, setting the `globalPackagesFolder` in *NuGet.Config* places the restored NuGet packages in the specified folder. This is an alternative to specifying the `--packages` option on the `dotnet restore` command. For more information, see the [NuGet.Config reference](/nuget/schema/nuget-config-file).

There are three specific settings that `dotnet restore` ignores.

* [bindingRedirects](/nuget/schema/nuget-config-file#bindingredirects-section)

  Binding redirects does not work with `<PackageReference>` elements and .NET Core only supports `<PackageReference>` elements for NuGet packages.

* [solution](/nuget/schema/nuget-config-file#solution-section)

>   This setting is only Visual Studio specific and no longer applies to the latest version of Visual Studio.

Reviewer: Is it worth/possible to mention the concrete version of Visual Studio, starting with which the setting doesn't apply anymore?

Author: I'll see if I can find that info. I was just told "it no longer applies"

Author: I clarified the real issue. "Legacy" was referring to the old project style that used packages.config. I just removed that wording and clarified what the difference really is.

> 

Reviewer 2: you'd probably want to update the ms.date here no?

Author: I generally don't update the date unless the entire article was refreshed/validated. I only updated the subsection area and didn't really look a the entire article. I figured this will be updated when the CLI updates come in for 3.0

For project-specific tooling, `dotnet restore` first restores the package in which the tool is packed, and then proceeds to restore the tool's dependencies as specified in its project file.

> ### Nuget.Config differences

Reviewer 2: I see the NuGet doc use different casing for this file. Is the file really capitalized like this or should it be nuget.config?

Author: Good catch. I didn't notice that, I think it should be `nuget.config` and no other weird casing. I think I was undoing the G in NuGet and didn't notice the C and really the N at the start.

The behavior of the `dotnet restore` command is affected by the settings in the *Nuget.Config* file, if present. For example, setting the `globalPackagesFolder` in *NuGet.Config* places the restored NuGet packages in the specified folder. This is an alternative to specifying the `--packages` option on the `dotnet restore` command. For more information, see the [NuGet.Config reference](/nuget/schema/nuget-config-file).

> There are three specific settings that `dotnet restore` ignores.

Reviewer 2: ```suggestion
There are three specific settings that `dotnet restore` ignores:
```

* [bindingRedirects](/nuget/schema/nuget-config-file#bindingredirects-section)

>   Binding redirects do not work with `<PackageReference>` elements and .NET Core only supports `<PackageReference>` elements for NuGet packages.

Reviewer 2: ```suggestion
  Binding redirects don't work with `<PackageReference>` elements and .NET Core only supports `<PackageReference>` elements for NuGet packages.
```

* [solution](/nuget/schema/nuget-config-file#solution-section)

>   This setting is only Visual Studio specific and does not apply to .NET Core. .NET Core does not use a `packages.config` file and instead uses `<PackageReference>` elements for NuGet packages.

Reviewer 2: ```suggestion
  This setting is Visual Studio specific and doesn't apply to .NET Core. .NET Core doesn't use a `packages.config` file and instead uses `<PackageReference>` elements for NuGet packages.
```

* [trustedSigners](/nuget/schema/nuget-config-file#trustedsigners-section)

>   This setting is not applicable as .NET Core does not yet support cross-platform verification of trusted packages.

Reviewer 3: Helpful if we link the tracking issue for cross platform verification?

Reviewer 3: This is the tracking issue: https://github.com/NuGet/Home/issues/7939

>   This setting is not applicable as .NET Core does not yet support cross-platform verification of trusted packages.

Reviewer 4: ```suggestion
  This setting is not applicable as [.NET Core does not yet support cross-platform verification](https://github.com/NuGet/Home/issues/7939) of trusted packages.
```

>   This setting is not applicable as [NuGet does not yet support cross-platform verification](https://github.com/NuGet/Home/issues/7939) of trusted packages.

Reviewer 2: ```suggestion
  This setting isn't applicable as [NuGet doesn't yet support cross-platform verification](https://github.com/NuGet/Home/issues/7939) of trusted packages.
```
